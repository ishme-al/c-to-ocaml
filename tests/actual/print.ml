[@@@ocaml.warning "-26"]
[@@@ocaml.warning "-27"]
[@@@ocaml.warning "-32"]
[@@@ocaml.warning "-69"]

open Core

let rec set_at_index (lst : 'a list) (index : int) (value : 'a) : 'a list =
  match lst with
  | [] -> failwith "Index out of bounds"
  | hd :: tl ->
      if index = 0 then value :: tl else hd :: set_at_index tl (index - 1) value

let foo () () = ()

let () =
  let x : int list = [ 1; 2; 3 ] in
  let y : float = 2. in
  printf "Hello, world %d %.2f!\n" (List.nth_exn x 1) y;
  exit 0
