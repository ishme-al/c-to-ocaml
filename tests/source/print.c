#include <stdio.h>

void foo() {
  return;
}

int main() {
    int x = 1;
    float y = 2.0;
    printf("Hello, world %d %.2f!\n", x, y);
    return 0;
}