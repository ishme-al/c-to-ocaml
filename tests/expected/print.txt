Hello, world 2 2.00!

